packages/cli/src/utils/generate.ts

```diff
@@ -0,0 +1,160 @@
+import fse from 'fs-extra'
+import fs from 'fs'
+import deepmerge from 'deepmerge'
+
+import {
+  CMSCoreDir,
+  CMSCustomDir,
+  CMSTmpDir,
+  coreDir,
+  customizationsDir,
+  customSrcDir,
+  repoDir,
+  storeConfigCoreFileDir,
+  storeConfigFileDir,
+  storeConfigTmpFileDir,
+  themeCustomizationsFileDir,
+  themesFileDir,
+  tmpDir,
+  tmpFolderName,
+} from './directory'
+
+interface GenerateOptions {
+  setup?: boolean
+}
+
+const ignoredDirs = ['node_modules']
+
+async function importStoreConfig() {
+  const { default: storeConfig } = await import(`${repoDir}/store.config.js`)
+  return storeConfig
+}
+
+function createTmpFolder() {
+  try {
+    fse.mkdirsSync(tmpDir)
+  } catch (err) {
+    console.error(err)
+  } finally {
+    console.log(`Temporary folder ${tmpFolderName} created`)
+  }
+}
+
+function copyCoreFiles() {
+  try {
+    fse.copySync(coreDir, tmpDir, {
+      filter(src) {
+        const splittedSrc = src.split('/')
+        return ignoredDirs.indexOf(splittedSrc[splittedSrc.length - 1]) === -1
+      },
+    })
+  } catch (e) {
+    console.error(e)
+  } finally {
+    console.log(`Core files copied`)
+  }
+}
+
+function copyCustomSrcToCustomizations() {
+  try {
+    fse.copySync(customSrcDir, customizationsDir)
+  } catch (err) {
+    console.error(err)
+  } finally {
+    console.log('Copied custom files')
+  }
+}
+
+async function copyTheme() {
+  const storeConfig = await importStoreConfig()
+
+  try {
+    fs.copyFileSync(
+      `${themesFileDir}/${storeConfig.theme}.scss`,
+      themeCustomizationsFileDir
+    )
+  } catch (err) {
+    console.error(err)
+  } finally {
+    console.log('Copied custom styles')
+  }
+}
+
+function mergeCMSFile(fileName: string) {
+  const coreContentTypes = fs.readFileSync(`${CMSCoreDir}/${fileName}`, 'utf8')
+  const customContentTypes = fs.readFileSync(
+    `${CMSCustomDir}/${fileName}`,
+    'utf8'
+  )
+  const coreContentTypesJSON = JSON.parse(coreContentTypes)
+  const customContentTypesJSON = JSON.parse(customContentTypes)
+
+  const mergeContentTypes = [...coreContentTypesJSON, ...customContentTypesJSON]
+
+  try {
+    fs.writeFileSync(
+      `${CMSTmpDir}/${fileName}`,
+      JSON.stringify(mergeContentTypes)
+    )
+  } catch (err) {
+    console.error(err)
+  } finally {
+    console.log(`CMS file ${fileName} created`)
+  }
+}
+
+function generateStoreConfigFile(content: any) {
+  return `module.exports = ${JSON.stringify(content)}\n`
```

Comment:
This will make the output look better :)

```suggestion
  return `module.exports = ${JSON.stringify(content, null, 2)}\n`
```